n3fit/src/n3fit/model_gen.py

```diff
@@ -750,17 +750,28 @@ def initializer_generator(seed, i_layer):
     # list_of_pdf_layers[d][r] is the layer at depth d for replica r
     list_of_pdf_layers = []
     for i_layer, (nodes_out, activation) in enumerate(zip(nodes_list, activations)):
-        layers = [
-            base_layer_selector(
+        if layer_type == "dense":
+            layers = base_layer_selector(
                 layer_type,
-                kernel_initializer=initializer_generator(replica_seed, i_layer),
+                replica_seeds=replica_seeds,
+                kernel_initializer=initializer_generator(0, i_layer),
```

Comment:
One question, since the seeds come in `replica_seeds` and the `initializer_generator` function is built specifically for this branch... wouldn't it be better to remove the extra input?

Reply:
I'm not sure what precisely you propose? Note that we need 3 pieces of information: `replica_seeds`, `i_layer` as this is added to the replica seeds, and the initializer class. Currently the last two are combined by setting `i_layer` as the "single_initializer"'s seed and extracting it to add to the replica seeds in `MultiInitializer`. How did you want to rewrite this?

Reply:
Sorry, I meant that I think `initializer_generator(0, i_layer)` could be defined simply as `initializer_generator(i_layer)` given that the function is defined before specifically for this branch of the conditional I believe.

Reply:
Thanks, makes sense. In fact looking at that section of code again it was unnecessarily complicated, I've simplified it a lot (and tested that results are still identical with 1 or 2 replicas, or 1 with the per flavour layer).

(I had to do the comparisons on snellius, as I'm having some bizarre issue after updating my mac, where two subsequent identical runs give different results. I'll open an issue about it.)